File: Ornek/src/app/admin/models/ITranslation.ts
import { ICategoryT } from './ICategoryT';
import { IContentT } from './IContentT';

export interface ITranslation
{
	ID : number,
	TransName : string,
	ShortName : string,
	Flag : string,
	Active : boolean,
	Deleted : boolean,

	Mesaj : string,

	CategoryTList : Array<ICategoryT>,
	ContentTList : Array<IContentT>,

	OldFlag: string,
	FlagHasFile : boolean,
}

File: Ornek/Repository/Translation/ITranslation.cs
using System;
using System.Collections.Generic;
using Repository.CategoryTModel;
using Repository.ContentTModel;

namespace Repository.TranslationModel
{
	public interface ITranslation
	{
		#region Model

		int ID { get; set; }
		string TransName { get; set; }
		string ShortName { get; set; }
		string Flag { get; set; }
		bool Active { get; set; }
		bool Deleted { get; set; }

		string Mesaj { get; set; }

		string OldFlag { get; set; }

		bool FlagHasFile { get; set; }

		List<ICategoryT> CategoryTList { get; set; }
		List<IContentT> ContentTList { get; set; }

		#endregion

		#region Methods

		List<Translation> List(int? id, int? top, bool relation);
		List<Translation> ListAll(int? id, bool relation);
		ITranslation Select(int? id, bool relation);
		ITranslation Insert(ITranslation table, bool? none);
		bool Insert(ITranslation table);
		ITranslation Update(int? id, ITranslation table);
		bool Update(ITranslation table);
		bool Copy(int id);
		bool Delete(int? id);
		bool Remove(int? id);

		#endregion
	}
}
